platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -2346,3 +2369,223 @@ def get_error_description(self):
         else:
             error_description = "Unknow SFP module status ({})".format(oper_status)
         return error_description
+
+
+class RJ45Port(SFP):
+    """class derived from SFP, representing RJ45 ports"""
+
+    def __init__(self, sfp_index):
+        super(RJ45Port, self).__init__(sfp_index, RJ45_TYPE)
+        self._sfp_type = RJ45_TYPE
+
+    # Reuse sdk_handle
+
+    @property
+    def sfp_type(self):
+        return self._sfp_type
+
+    def get_presence(self):
+        """
+        Retrieves the presence of the device
+
+        Returns:
+            bool: True if device is present, False if not
+        """
+        # Reimplement it via using SDK API
+        return True
+
+    @property
+    def dom_supported(self):
+        return False
+
+    def get_transceiver_info(self):
+        """
+        Retrieves transceiver info of this SFP
+
+        Returns:
+            A dict which contains following keys/values :
+        ================================================================================
+        keys                       |Value Format   |Information
+        ---------------------------|---------------|----------------------------
+        type                       |1*255VCHAR     |type of SFP
+        vendor_rev                 |1*255VCHAR     |vendor revision of SFP
+        serial                     |1*255VCHAR     |serial number of the SFP
+        manufacturer               |1*255VCHAR     |SFP vendor name
+        model                      |1*255VCHAR     |SFP model name
+        connector                  |1*255VCHAR     |connector information
+        encoding                   |1*255VCHAR     |encoding information
+        ext_identifier             |1*255VCHAR     |extend identifier
+        ext_rateselect_compliance  |1*255VCHAR     |extended rateSelect compliance
+        cable_length               |INT            |cable length in m
+        mominal_bit_rate           |INT            |nominal bit rate by 100Mbs
+        specification_compliance   |1*255VCHAR     |specification compliance
+        vendor_date                |1*255VCHAR     |vendor date
+        vendor_oui                 |1*255VCHAR     |vendor OUI
+        application_advertisement  |1*255VCHAR     |supported applications advertisement
+        ================================================================================
+        """
+        transceiver_info_dict = {}
+
+        transceiver_info_dict['type'] = self.sfp_type
```

Comment:
this means we support none of the items. is it correct to assume ALL are N/A beside of the type which i assume in this it is RJ45 only?

Reply:
Yes.
The dict should contain values for all fields. In case some fields are not supported, `N/A` should be returned.